File: tippay/apps/ios/FliqIOS/Sources/AuthModels.swift
import Foundation

enum NativeRole: String, CaseIterable, Identifiable {
    case customer = "CUSTOMER"
    case provider = "PROVIDER"
    case business = "BUSINESS"

    var id: String { rawValue }

    var title: String {
        switch self {
        case .customer: return "Customer"
        case .provider: return "Provider"
        case .business: return "Business"
        }
    }

    var credentialLabel: String {
        switch self {
        case .business: return "Email"
        case .customer, .provider: return "Phone"
        }
    }

    var usesEmail: Bool { self == .business }
}

struct AuthUser: Codable {
    let id: String
    let phone: String?
    let email: String?
    let name: String?
    let type: String
    let kycStatus: String?
    let languagePreference: String?
}

struct AuthSession: Codable {
    let accessToken: String
    let refreshToken: String
    let user: AuthUser
}

struct SendCodeResult: Decodable {
    let message: String
    let otp: String?
}

struct VerifyCodeResponse: Decodable {
    let accessToken: String
    let refreshToken: String
    let user: AuthUser
}

struct APIMessageResponse: Decodable {
    let message: String?
}

File: tippay/apps/android/app/src/main/java/com/fliq/android/data/AuthModels.kt
package com.fliq.android.data

import org.json.JSONObject

enum class NativeRole {
    CUSTOMER,
    PROVIDER,
    BUSINESS,
}

data class AuthUser(
    val id: String,
    val phone: String?,
    val email: String?,
    val name: String?,
    val type: String,
    val kycStatus: String?,
    val languagePreference: String?,
) {
    companion object {
        fun fromJson(json: JSONObject): AuthUser {
            return AuthUser(
                id = json.getString("id"),
                phone = json.optString("phone").takeIf { it.isNotBlank() },
                email = json.optString("email").takeIf { it.isNotBlank() },
                name = json.optString("name").takeIf { it.isNotBlank() },
                type = json.optString("type"),
                kycStatus = json.optString("kycStatus").takeIf { it.isNotBlank() },
                languagePreference = json.optString("languagePreference").takeIf { it.isNotBlank() },
            )
        }
    }

    fun toJson(): JSONObject {
        return JSONObject().apply {
            put("id", id)
            put("phone", phone)
            put("email", email)
            put("name", name)
            put("type", type)
            put("kycStatus", kycStatus)
            put("languagePreference", languagePreference)
        }
    }
}

data class AuthSession(
    val accessToken: String,
    val refreshToken: String,
    val user: AuthUser,
)

data class SendCodeResult(
    val message: String,
    val otpHint: String?,
)

class AuthException(
    override val message: String,
    val statusCode: Int,
) : Exception(message)
